velox/functions/sparksql/Arithmetic.h

```diff
@@ -262,6 +262,17 @@ struct Log1pFunction {
   }
 };
 
+template <typename T>
+struct LogarithmFunction {
+  FOLLY_ALWAYS_INLINE bool call(double& result, double a, double b) {
```

Comment:
Could we check if this implementation gives consistent result with Spark when there is special value like NaN, Inf and -Inf? Thanks.

Reply:
UTs are added, thanks.
![image](https://github.com/facebookincubator/velox/assets/10524738/ebd341ed-5ca8-4d17-8a2a-05cda94b36e5)

Reply:
Thanks for checking. Could we add test for log(inf, inf)?
>spark.sql("select log(cast('Inf' as double), cast('Inf' as double))").show(false)
+---------------------------------------------+
|LOG(CAST(Inf AS DOUBLE), CAST(Inf AS DOUBLE))|
+---------------------------------------------+
|NaN                                          |
+---------------------------------------------+

Reply:
Added ,thanks.